line_profiler/line_profiler.py

```diff
@@ -25,6 +19,13 @@
 __version__ = '3.5.0'
 
 
+def load_ipython_extension(ip):
+    """ API for IPython to recognize this module as an IPython extension.
+    """
+    from .ipython_extension import LineProfilerMagics
```

Comment:
It can either be imported from this function, or it can be added with the other imports with a sys.modules guard for IPython.
In the second case there would be a circular import but it shouldn't be a problem.